app/src/pages/profile/APIKeysTable.tsx

```diff
@@ -0,0 +1,209 @@
+import React, { startTransition, useCallback, useMemo } from "react";
+import { graphql, useMutation, useRefetchableFragment } from "react-relay";
+import {
+  ColumnDef,
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+
+import { Flex, Icon, Icons } from "@arizeai/components";
+
+import { DeleteAPIKeyButton } from "@phoenix/components/auth";
+import { TextCell } from "@phoenix/components/table";
+import { tableCSS } from "@phoenix/components/table/styles";
+import { TableEmpty } from "@phoenix/components/table/TableEmpty";
+import { TimestampCell } from "@phoenix/components/table/TimestampCell";
+import { useNotifySuccess } from "@phoenix/contexts";
+
+import { APIKeysTableFragment$key } from "./__generated__/APIKeysTableFragment.graphql";
+import { APIKeysTableQuery } from "./__generated__/APIKeysTableQuery.graphql";
+
+const TIMESTAMP_CELL_SIZE = 70;
+
+export function APIKeysTable({ query }: { query: APIKeysTableFragment$key }) {
+  const [data, refetch] = useRefetchableFragment<
+    APIKeysTableQuery,
+    APIKeysTableFragment$key
+  >(
+    graphql`
+      fragment APIKeysTableFragment on User
+      @refetchable(queryName: "APIKeysTableQuery") {
+        apiKeys {
+          id
+          name
+          description
+          createdAt
+          expiresAt
+        }
+      }
+    `,
+    query
+  );
+
+  const notifySuccess = useNotifySuccess();
+  const [commit] = useMutation(graphql`
+    mutation APIKeysTableDeleteAPIKeyMutation($input: DeleteApiKeyInput!) {
+      deleteUserApiKey(input: $input) {
+        __typename
+        apiKeyId
+        query {
+          ...UserAPIKeysTableFragment
+        }
```

Comment:
this is an attempt at making the userapi keys table (on settings page) refetch - it doesn't work however so i changed the fetch policy there, can remove since it's not working